### PR TITLE
Molecule upgrade & idempotence fix

### PR DESCRIPTION
**Pull request by sbesson**
Driven by https://github.com/IDR/deployment/pull/397, this PR:

- upgrades the testing infrastructure to Molecule 3+
- fixes the idempotence issue by annotating the tasks using the `PUBLIC ` role - see https://docs.ansible.com/ansible/latest/collections/community/postgresql/postgresql_privs_module.html
- remove EOL PSQL 10 and add PSQL 14 to the testing matrix

### CODE REVIEW

File: .github/workflows/molecule.yml [hunk 1]
```diff
@@ -29,10 +29,16 @@ jobs:
       matrix:
         scenario: ${{fromJson(needs.list-scenarios.outputs.matrix)}}
     steps:
-      - uses: actions/checkout@v2
-      - uses: ome/action-ome-ansible-molecule@main
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
```
Comment (jburel) [inline .github/workflows/molecule.yml:33]: Is the idea to remove the usage of ``action-ome-ansible-molecule`` and replace  it by the code below everywhere? The original action used could be adjusted to cover the new workflow.

Comment (sbesson) [reply .github/workflows/molecule.yml:33]: Discussed briefly with @jmarie - see also the description of https://github.com/IDR/deployment/pull/397. I will push additional commits to these PRs to skip the `action-ome-ansible-molecule` action or the `ome-ansible-molecule` meta-packages and open an issue to discuss a possible deprecation of these repositories.

**Comment by sbesson** [comment]
The last commit also upgrades the version of Ansible used for testing this role to the current stable release 7.x which requires Python 3.9 inline with the decisions made in https://github.com/IDR/deployment/pull/397

From my side this is now ready for review.

**Comment by sbesson** [comment]
@jburel @joshmoore I keep getting notifications of failed Actions on this repo and other Ansible ones. Do you have a roadmap on how you want to proceed ?

**Comment by pwalczysko** [comment]
Do we want to remove the Ubuntu 18.04 testing here too ? (similarly to the strategy in ansible-role-omero-web and basedeps) https://github.com/ome/ansible-role-postgresql/blob/51fc1e0718cce7f4d49dd6dcffbaf2e1e078e080/.github/workflows/molecule.yml#L25

**Comment by sbesson** [comment]
Thanks @pwalczysko for the suggestion and agreed. Pushed a commit removing testing (& support) for EOL 18.04

**Review by pwalczysko — APPROVED** [review]

**Pull request merged**